### PR TITLE
[MPLUGINTESTING-76] Upgrade Plexus Archiver from 4.3.0 to 4.6.1 (broken api in 4.4.0)

### PR DESCRIPTION
**Pull request by gnodet**

### CODE REVIEW

**Review by michael-o — APPROVED** [review]

File: maven-plugin-testing-harness/src/main/java/org/apache/maven/plugin/testing/ArtifactStubFactory.java [hunk 1]
```diff
@@ -273,17 +273,9 @@ public void createUnpackableFile(Artifact artifact, File destFile)
         archiver.setDestFile(destFile);
         archiver.addFile(srcFile, getUnpackableFileName(artifact));
 
-        try {
-            setVariableValueToObject(archiver, "logger", new SilentLog());
```
Comment (slawekjaranowski) [inline maven-plugin-testing-harness/src/main/java/org/apache/maven/plugin/testing/ArtifactStubFactory.java:277]: Looks like `SilentLog` class is no more used ... maybe we can remove it, or at least deprecate

**Review by slawekjaranowski — APPROVED** [review]

File: maven-plugin-testing-harness/src/main/java/org/apache/maven/plugin/testing/ArtifactStubFactory.java (hunk 1 above)
Comment (cstamas) [reply maven-plugin-testing-harness/src/main/java/org/apache/maven/plugin/testing/ArtifactStubFactory.java:277]: If it is in any relation to Plexus Logger, then definitely

Comment (gnodet) [reply maven-plugin-testing-harness/src/main/java/org/apache/maven/plugin/testing/ArtifactStubFactory.java:277]: It's used in various places [within the ASF](https://github.com/search?q=org%3Aapache+org.apache.maven.plugin.testing.SilentLog&type=code&p=2) and there's no real replacement.  While maven components should be switched to SLF4j, the plugin side is not completely sorted (as afaik, the `org.slf4j` package is not exported by the maven core class loader.
So I've raised [an issue](https://issues.apache.org/jira/browse/MPLUGINTESTING-83) but I'd keep it as is until it's sorted.

Comment (slawekjaranowski) [reply maven-plugin-testing-harness/src/main/java/org/apache/maven/plugin/testing/ArtifactStubFactory.java:277]: `org.slf4j` is exported 

https://github.com/apache/maven/blob/master/maven-core/src/main/resources/META-INF/maven/extension.xml#L128-L132

**Pull request merged**